diff -ur a/lib/config.guess b/lib/config.guess
--- a/lib/config.guess	2017-06-18 10:37:14.000000000 +0800
+++ b/lib/config.guess	2021-07-27 19:58:33.842072143 +0800
@@ -976,6 +976,9 @@
     k1om:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-${LIBC}
 	exit ;;
+    loongarch32:Linux:*:* | loongarch64:Linux:*:* | loongarchx32:Linux:*:*)
+        echo "$UNAME_MACHINE"-unknown-linux-"$LIBC"
+        exit ;;
     m32r*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-${LIBC}
 	exit ;;
diff -ur a/lib/config.sub b/lib/config.sub
--- a/lib/config.sub	2017-06-18 10:37:14.000000000 +0800
+++ b/lib/config.sub	2021-07-27 20:01:01.410343521 +0800
@@ -268,6 +268,7 @@
 	| k1om \
 	| le32 | le64 \
 	| lm32 \
+        | loongarch32 | loongarch64 | loongarchx32 \
 	| m32c | m32r | m32rle | m68000 | m68k | m88k \
 	| maxq | mb | microblaze | microblazeel | mcore | mep | metag \
 	| mips | mipsbe | mipseb | mipsel | mipsle \
